pkg/cluster/internal/create/actions/kubeadmjoin/join.go

```diff
@@ -211,6 +212,16 @@ func runKubeadmJoinControlPlane(
 		return errors.Wrap(err, "failed to join a control plane node with kubeadm")
 	}
 
+	// Wait for the node to be Ready
+	// TODO: remove once https://github.com/kubernetes-sigs/kind/issues/588 is fixed
+	// kubeadm join should guarantee that the cluster is ready
+	startTime := time.Now()
+	// TODO: currently hardcoded to 10s
+	isReady := nodes.WaitForReady(node, startTime.Add(10*time.Second))
```

Comment:
Super new to the code so I could be completely off, but isn't there a [`WaitForReady`](https://godoc.org/sigs.k8s.io/kind/pkg/cluster/create#WaitForReady)  option already that could be used here?

Reply:
yeah, indeed this function `nodes.WaitForReady` is used by the one you mentioned.
In this case we are looking for a specific node so I think that's better to use this one,